File: 9-palindrome-number/palindrome-number.java
public class Solution {
    public boolean isPalindrome(int x) {
        int n = x;
        long rev = 0;
        while (x > 0) {
            rev = (rev * 10) + (x % 10);
            x /= 10;
        }
        if (rev == n) return true;
        return false;
    }
}

File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        n = x
        rev = 0
        if x < 0:
            return False
        while x > 0:
            rev = (rev * 10) + (x % 10)
            x //= 10
        if rev == n:
            return True
        return False